File: leetcode/python/1491_average_salary_excluding_minimum_and_maximum_salary.py
class Solution:
    def average(self, salary: list[int]) -> float:
        total = curMax = 0
        curMin = 1e6
        for n in salary:
            total += n
            curMax = max(curMax, n)
            curMin = min(curMin, n)
        return (total - curMax - curMin) / (len(salary) - 2)

File: leetcode/rust/src/1491_average_salary_excluding_minimum_and_maximum_salary.rs
use std::cmp;

impl Solution {
    pub fn average(salary: Vec<i32>) -> f64 {
        let mut min = salary[0];
        let mut max = salary[0];
        let mut sum = 0;
        for s in &salary {
            sum += *s;
            min = cmp::min(min, *s);
            max = cmp::max(max, *s);
        }
        (sum - min - max) as f64 / (salary.len() - 2) as f64
    }
}